### ISSUE TITLE
CustomTabs working on IOS but not on Android. And  remounts components/screens when change tab.

### ISSUE THREAD

**Issue by MichelDiz** [react-navigation/react-navigation#4038]
### Current Behavior

- What code are you running and what is happening?

 I have copied part of a code from Spencer. About Auth Flow. All works nice before my custom SignedIn stack made as:

```Typescript
const CustomTabs = createNavigationContainer(
  createNavigator(CustomTabRouter)(CustomTabView as any),  // ! I'm using TypeScript
);
```
```Typescript
export const FeedStack = StackNavigator({
  Feed: {
	screen: Home,
	path: "HomeFeed",
	},
  Details: {
	screen: Details,
	path: "Details",
},
},
{
	initialRouteName: "Feed",
	mode: "modal",
	headerMode: "none",
},
);
```

```Typescript
export const CustomTabRouter = TabRouter(
  {
	Home: {
		screen: FeedStack,
		path: "Home",
	},
	FAQ: {
		screen: FAQ,
		path: "FAQ",
	},
	Settings: {
		screen: Settings,
		path: "settings",
	},
	Carteira: {
		screen: Carteira,
		path: "Carteira",
	},
  },
  {
	initialRouteName: "Home",
  },
);
```

```Typescript
const CustomTabView = ({ router, navigation }) => {
  const { routes, index } = navigation.state;
	const ActiveScreen = router.getComponentForState(navigation.state);
  return (
	<View style={styles.container}>
	<ScrollView>
		<ActiveScreen style={{ flex: 1 }}
		navigation={addNavigationHelpers({
			...navigation,
			state: routes[index],
		})}
		/>
		</ScrollView>
		<CustomTabBar navigation={navigation} />
	</View>
  );
};

```

```Typescript
export const createRootNavigator = (signedIn = false) => {
  console.log(signedIn, "signedIn ?");
  return SwitchNavigator(
	{
		SignedIn: {
		screen: CustomTabs, // ! This section re-render all screens between tabs and sub stacks(I did a test)
		},
		SignedOut: {
		screen: SignedOut, // ! This section works great, as expected.
		},
	},
	{
		initialRouteName: signedIn ? "SignedIn" : "SignedOut",
	},
  );
};

```

I just wanted to create a custom Tab. And it's very good on IOS, I was able to create as I wanted from design. I came across this two problems only now using createNavigationContainer for CustomTabs. 

Video from the action https://youtu.be/Fe5uW_I202E

![captura de tela 2018-04-22 as 21 41 22](https://user-images.githubusercontent.com/6570955/39102646-3496e5e8-467c-11e8-9a40-953fceaa87ed.png)

### Expected Behavior

- What do you expect should be happening?

Do not reassemble my screens as if it were a "page refresh". Or forget the previous state. You can replicate this behavior even in the example in the Samples folder. I tested it at the Expo and it actually refreshes the screen.

Despite this I can only display my custom Tab in IOS. Android does not render anything. By the way I'm using Typescript.

### How to reproduce

1 - You can test with the same example provide in Examples NavigationPlayground. In the issue of the screen updating unnecessarily.
2 - The only changes I made to the code were as shown above, Besides that I used an SVG for the background. And I used the "NativeBase" footer to keep it at the bottom of the screens. In addition to by in "absolute" position in the style.

### Your Environment

| software         | version
| ---------------- | -------
| react-navigation | "react-navigation": "^1.5.11"
| react-native     | "react-native": "^0.55.3",
| node             | v9.3.0
| npm or yarn      | npm -v 5.5.1
| npm or yarn      | yarn -v 1.5.1

**Comment by brentvatne** [comment]
you have not provided a runnable example as per the template. please do so in a new issue.

**Comment by MichelDiz** [comment]
Did you even readed it?

**Comment by MichelDiz** [comment]
I took a time testing, reading a significant amount of issues of this project. To no have to open a new issue ...
It took me a while to fill in everything I needed. To simply have such an answer? Wow.

In "How to reproduce" there is how to "reproduce" the problem. Actually this "problem" is a "feature" ...

The second problem I think is something with Android. But no need to worry, I gave up using React-navigation this way.

Cheers
